### PR TITLE
COCOS-153 - Add `host-data` option

### LINKED ISSUE #153: Feature: Add `host-data` QEMU option to CVM boot up
### Is your feature request related to a problem? Please describe.

The idea is to provide the Agent with a way to check what he receives from the Manager and to provide the consortium members with a way to check if the CVM they are communicating with is indeed their CVM.

### Describe the feature you are requesting, as well as the possible use case(s) for it.

This feature aims to add a `host-data` option to the QEMU launch command. This option introduces a 256-bit value that will become a permanent field in the attestation report (AR). 

With the `host-data` option, the Manager can launch the CVM and pass the hash of the manifest as the value of this option. The Agent can then use this field of the AR to verify the authenticity of the manifest received from the Manager. 

During the AR verification process, the consortium members can verify the AR's `host-data` field and assure themselves that the CVM they are communicating with is indeed the expected one.

### Indicate the importance of this feature to you.

Must-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by danko-miladinovic**
# What type of PR is this?

This is a feature because it adds:
- QEMU `host-data` option to the launching of CVMs.

This PR also renames the `platform_info.json` into `backend_info.json`.
## What does this do?
This PR enables the Manager to launch CVMs with the `host-data` filed of the attestation report populated with the computation configuration of the Agent. This way, the Agent can check what is being sent to him by the Manager.
This PR also renames the `platform_info.json` into `backend_info.json`.

## Which issue(s) does this PR fix/relate to?

- Resolves #153  

## Have you included tests for your changes?

The manual README.md has been changed to reflect this new feature.
## Did you document any new/modified feature?

Documentation will be updated in a separate PR.

### CODE REVIEW

File: manager/service.go [hunk 1]
```diff
@@ -104,6 +109,16 @@ func (ms *managerService) Run(ctx context.Context, c *manager.ComputationRunReq)
 	}
 	ms.qemuCfg.HostFwdAgent = agentPort
 
+	jsonData, err := json.Marshal(ac)
+	if err != nil {
+		ms.publishEvent("vm-provision", c.Id, "failed", json.RawMessage{})
+		return "", errors.Wrap(ErrFailedToMarshalJSON, err)
+	}
+	computationHash := sha3.Sum256(jsonData)
```
[the remark this answers is not in the document]
Comment (danko-miladinovic) [reply manager/service.go]: Done

File: manager/service.go [hunk 2]
```diff
@@ -156,3 +170,12 @@ func (ms *managerService) publishEvent(event, cmpID, status string, details json
 		},
 	}
 }
+
+func getComputationHash(ac *agent.Computation) ([32]byte, error) {
```
Comment (dborovcanin) [inline manager/service.go:174]: Rename to `computationHash` and use like `ch := computationHash(ac)`. If there is no need to pass the pointer param here (i.e. we do not change the computation from this function), pass by value instead.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: cli/backend_info.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+const (
+	filePermision     = 0o755
+	hostDataLength    = 32
+	measurementLength = 48
+)
+
+type AttestationConfiguration struct {
+	SNPPolicy   *check.Policy      `json:"snp_policy,omitempty"`
+	RootOFTrust *check.RootOfTrust `json:"root_of_trust,omitempty"`
```
[the remark this answers is not in the document]
Comment (danko-miladinovic) [reply cli/backend_info.go]: Done, my mistake.

File: manager/service.go (hunk 2 above)
Comment (danko-miladinovic) [reply manager/service.go]: Done. The computation is passed by value now.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: cli/backend_info.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+				log.Fatalf("Error could not truncate backend information JSON file: %v", err)
+			}
+
+			fileJson, err := json.MarshalIndent(attestationConfiguration, "", " ")
+			if err != nil {
+				log.Fatalf("Error marshaling the backend information JSON: %v", err)
+			}
+			if err = os.WriteFile(backendInfo.Name(), fileJson, filePermision); err != nil {
+				log.Fatalf("Error writing into backend information JSON file: %v", err)
+			}
```
[the remark this answers is not in the document]
Comment (danko-miladinovic) [reply cli/backend_info.go]: Done

File: cli/backend_info.go [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+			fmt.Printf("\nFlags:\n")
+			cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+				fmt.Printf("  -%s, --%s %s\n", flag.Shorthand, flag.Name, flag.Usage)
+			})
+			fmt.Printf("\nUse \"%s [command] --help\" for more information about a command.\n", cmd.CommandPath())
+		},
+	}
+}
+
+func (cli *CLI) NewAddMeasurementCmd() *cobra.Command {
```
Comment (dborovcanin) [inline cli/backend_info.go:59]: There are a lot of repetitions, please extract to an unexported function.

File: cli/backend_info.go [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+	"log"
+	"os"
+
+	"github.com/google/go-sev-guest/proto/check"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const (
+	// 0o744 file permission gives RWX permission to the user and only the R permission to others
```
Comment (SammyOina) [inline cli/backend_info.go:18]:
```suggestion
	// 0o744 file permission gives RWX permission to the user and only the R permission to others.
```

**Review by SammyOina — CHANGES_REQUESTED** [review]

File: cli/backend_info.go (hunk 5 above)
Comment (danko-miladinovic) [reply cli/backend_info.go:77]: Changed

**Review by SammyOina — APPROVED** [review]

File: cli/backend_info.go [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+	if err = json.Unmarshal(backendInfo, &ac); err != nil {
+		return errors.Wrap(errUnmarshalJSON, err)
+	}
+
+	switch field {
+	case measurementField:
+		ac.SNPPolicy.Measurement = data
+	case hostDataField:
+		ac.SNPPolicy.HostData = data
```
Comment (dborovcanin) [inline cli/backend_info.go:130]: Add a default case that returns an error.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**